### PR TITLE
Support Typescript / ember-cli-typescript

### PR DESCRIPTION
**Pull request by lifeart**

### CODE REVIEW

**Comment by rwwagner90** [comment]
@lifeart this would close #212 right? Seems good to me

**Comment by lifeart** [comment]
@rwwagner90 yes :)
// cc @makepanic, @NullVoxPopuli, @luketheobscure, @mdeanjones, @gmurphey, @scottkidder, @kategengler

**Comment by NullVoxPopuli** [comment]
This does not work for me. Typescript files are ignored. :(

image

**Comment by lifeart** [comment]
@NullVoxPopuli, plz retry) small typo fixed

**Comment by NullVoxPopuli** [comment]
that did it!
image

File: index.js [hunk 1]
```diff
@@ -196,12 +196,14 @@ module.exports = {
    * @returns {Array<String>} include paths
    */
   _getIncludesForDir: function(dir, prefix) {
+    const hasEmberCliTypescript = this.project && this.projectFindAddonByName && this.project.findAddonByName('ember-cli-typescript');
```
Comment (makepanic) [inline index.js]: this seems like a typo?

Shouldn't the second check be `this.project.findAddonByName`?

Comment (lifeart) [reply index.js]: @makepanic nice catch! Fixed

Comment (makepanic) [reply index.js]: Oops didn't see that you've already fixed it. Had the same issue nullvox initially reported and saw that in a diff between the other PR.

**Comment by NullVoxPopuli** [comment]
I'm confused how some of these are untested : https://codeclimate.com/github/NullVoxPopuli/emberclear/code?sort=test_coverage

Resolver is used by ember itself.
Logout is for sure tested. :/
Am confused

**Comment by lifeart** [comment]
image

**Comment by makepanic** [comment]
Can confirm that with the fixed typo `ts` coverage works again.

**Comment by lifeart** [comment]
@NullVoxPopuli it can be `pods` issue (don't related to `.ts` itselft) 
I can't click on uncovered file link (404)

**Comment by NullVoxPopuli** [comment]
Yeah, I think I need to update how I upload the coverage since I also use a monorepo

File: config/ember-try.js [hunk 2]
```diff
@@ -1,80 +1,76 @@
...
-const getChannelURL = require('ember-source-channel-url');
+const getChannelURL = require("ember-source-channel-url");
 
 module.exports = function() {
   return Promise.all([
-    getChannelURL('release'),
-    getChannelURL('beta'),
-    getChannelURL('canary')
-  ]).then((urls) => {
+    getChannelURL("release"),
```
Comment (rwwagner90) [inline config/ember-try.js]: @lifeart were all these quotes changed to double quotes on purpose? I thought the blueprint used single quotes?

Comment (lifeart) [reply config/ember-try.js]: @rwjblue it can be an `eslint` or `prettier` auto format rule

File: test/unit/index-test.js [hunk 3]
```diff
@@ -496,11 +496,11 @@ describe('index.js', function() {
         it('instruments the inrepo addon', function() {
           const includes = Index._getIncludesForInRepoAddonDirectories();
           expect(includes).to.deep.equal([
-            'my-app/utils/my-covered-util.js',
```
Comment (rwwagner90) [inline test/unit/index-test.js]: Do we no longer support folders for the app, in-repo-addon, etc? I think we should, right?

File: config/ember-try.js (hunk 2 above)
Comment (rwwagner90) [reply config/ember-try.js]: We should use whatever the default blueprint format is. I think it is single quote in all the Ember repos.

File: test/unit/index-test.js (hunk 3 above)
Comment (lifeart) [reply test/unit/index-test.js]: have no idea(
This changes just to get tests passed

Comment (rwwagner90) [reply test/unit/index-test.js]: Hmm, maybe something changed in the resolver or something. I don't think we want to change the paths though.

Comment (lifeart) [reply test/unit/index-test.js]: how we can fix tests in this case?

Comment (makepanic) [reply test/unit/index-test.js]: I tried to investigate this change and it seems like the test is green if i run `yarn node-test` locally in `code-coverage-upstream-oct-2019` with changes in `test/unit/index-test.js` reverted :thinking: 

It even fails if i keep the changed index-test.js 



I tried this in both node8 and node latest

Comment (lifeart) [reply test/unit/index-test.js]: I suppose it was failing on my windows machine, anyway changes reverted

File: package.json [hunk 4]
```diff
@@ -22,65 +22,67 @@
...
     "loader.js": "^4.7.0",
     "mocha": "^5.2.0",
-    "qunit-dom": "^0.7.1",
+    "qunit-dom": "^0.8.4",
     "rimraf": "^2.6.2",
     "sinon": "^4.2.2"
   },
   "engines": {
-    "node": "6.* || 8.* || >= 10.*"
+    "node": "8.* || >= 10.*"
```
Comment (Turbo87) [inline package.json]: FYI this should be considered a breaking change

File: config/ember-try.js [hunk 5]
```diff
@@ -7,31 +7,23 @@ module.exports = function() {
...
-          name: 'ember-lts-2.18',
-          env: {
-            EMBER_OPTIONAL_FEATURES: JSON.stringify({ 'jquery-integration': true }),
-          },
+          name: 'ember-lts-3.8',
           npm: {
             devDependencies: {
-              '@ember/jquery': '^0.5.1',
-              'ember-source': '~2.18.0'
+              'ember-source': '~3.8.0'
```
Comment (Turbo87) [inline config/ember-try.js]: same for this, it looks like we're dropping support for old Ember versions

**Pull request merged**